src/Fixer/PhpTag/BlankLineAfterOpeningTagFixer.php

```diff
@@ -86,6 +81,7 @@ public function getPriority()
      */
     public function isCandidate(Tokens $tokens)
     {
-        return $tokens->isTokenKindFound(T_OPEN_TAG);
+        // ignore files with short open tag and ignore non-monolithic files
+        return $tokens[0]->isGivenKind(T_OPEN_TAG) && $tokens->isMonolithicPhp();
```

Comment:
2nd thing, `->fix` method must still works properly if one forgot to call `->isCandidate`